src/server/server_family.h

```diff
@@ -358,6 +361,12 @@ class ServerFamily {
   std::unique_ptr<util::fb2::FiberQueueThreadPool> fq_threadpool_;
   std::shared_ptr<detail::SnapshotStorage> snapshot_storage_;
 
+  std::atomic<bool> client_pause_ = false;
+  // We need this because if dragonfly shuts down during pause, ServerState will destruct
+  // before the dettached fiber Pause() causing a seg fault.
+  std::atomic<size_t> active_pauses_ = 0;
+  util::fb2::EventCount client_pause_ec_;
```

Comment:
We might have the same bug in other calls to Pause() or basically any `detached fiber` that potentially accesses destructed objects during shutdown. I will take a look, but on a separate PR.

//TODO file an issue